test/single_file/assign_null.c

```diff
@@ -0,0 +1,3 @@
+int main() {
+    int* x = 0;
```

Comment:
Looks like in C it is acceptable to replace a null pointer constant with the result of a function call. So we *could* make the new behaviour conditional on C++ and have a test to ensure that for C the mutation still occurs. I'm in two minds about this. On balance though I think let's keep the stricter checking that you have implemented (i.e. no action needed).

Reply:
Sounds good, I'll add a note in the comment for the changed code explaining that it would be safe to enable for C as a reminder and in case we decide to change it in the future.